ocamltest/filecompare.ml

```diff
@@ -69,15 +68,56 @@ let read_file bytes_to_ignore filetype fn =
     | s -> loop (s :: acc)
     | exception End_of_file ->
       close_in ic;
-      try
-        if filetype = Text then
-          List.rev_map drop_cr acc
-        else
-          raise Exit
+      try List.rev_map drop_cr acc
       with Exit -> List.rev acc
   in
   loop []
 
+let compare_text_files file1 file2 =
+  if read_text_file file1 = read_text_file file2 then
+    Same
+  else
+    Different
+
+(* Version of Pervasives.really_input which stops at EOF, rather than raising
+   an exception. *)
+let really_input_up_to ic =
+  let block_size = 8192 in
+  let buf = Bytes.create block_size in
+  let rec read pos =
+    let bytes_read = input ic buf pos (block_size - pos) in
+    if bytes_read = 0 || bytes_read = (block_size - pos) then
+      block_size - pos - bytes_read
+    else
+      read (pos + bytes_read)
+  in
+  let bytes_read = block_size - read 0 in
```

Comment:
I find it rather strange that `read` returns the number of bytes **not** read, and this double-subtraction business is a bit hard to read. Why not just return `pos + bytes_read` at line 90, and replace line 94 with `let bytes_read = read 0` ?

I think this would be even more natural:
```
  let rec read pos =
    let bytes_read = input ic buf pos (block_size - pos) in
    let newpos = pos + bytes_read in
    if bytes_read = 0 || newpos = blocksize then
      newpos
    else
      read newpos
  in
  let bytes_read = read 0 in
```